clients/python-wrapper/tests/integration/test_reference.py

```diff
@@ -47,6 +50,19 @@ def test_reference_merge_into(setup_branch_with_commits):
     repo = lakefs.Repository(branch.repo_id)
     main = repo.branch("main")
 
+    # test merging into same branch
+
+    other_but_same_branch = repo.branch("test_reference_merge_into_no_changes").create(branch)
+    changes = list(branch.diff(other_but_same_branch))
+    assert len(changes) == 0
+
+    with pytest.raises(BadRequestException, match=r'.+no changes.+'):
+        branch.merge_into(other_but_same_branch, message="MergeNoChanges")
+
+    branch.merge_into(other_but_same_branch, message="MergeNoChangesWithFlag", allow_empty=True)
```

Comment:
This is definitely a bug and should not be allowed.
Can you please open an issue for it?